### ISSUE TITLE
Extent the Scale migration feature to allow switching ScaleTypes and DefaultScales!

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/TweakScale#236]
As suggested by myself on [Forum](https://forum.kerbalspaceprogram.com/index.php?/topic/182700-111x-all-tweak-07-23rdoctober2019/&do=findComment&comment=4102706).

Since I have access to the current craft values immediately after the OnLoad (as the _KSP Upgrade Pipeline_ thingy only acts after it), I can try to work out this task on `OnLoad` by mangling the config data myself, before calling `Base.OnLoad`.

**Comment by Lisias** [comment]
One problem of the original code implemented originally is that it was ignoring SaveGames and upgrading only craft files being loaded on the Editor. At that time, I was insecure about meddling with savegames, when a change of mine could ruin something - but as time gone by, my overzealously probably ended up being counter-productive...

So the new code will also handle opening ongoing savegames.

This code is probably over-engineered (the original could be easily adapted), mathematically it's not too different from the previous one - but, and again, I always choose the conservative way out of a problem and doing this way I can handle unseen use-cases easier.

The previous code is field proven anyway, I prefer not to take risks when savegames are involved.

**Comment by Lisias** [comment]
Something was smelling funny and while sniffing around, I came to conclusion that the original code was not working anymore - at least, on KSP 1.9.1.

I have the remembrance of testing that stunt, so I'm guessing some recent KSP release had broken my legs again. (sigh).

In a way or another, I decided to brute force my way into my own data - worst case, older KSP releases where the thingy used to work (or at least it's how I remember it) will have the work redone and that's it.

Code is tested and it's working now, tested it by (ab)using [All Tweak](https://forum.kerbalspaceprogram.com/index.php?/topic/182700-111x-all-tweak-07-23rdoctober2019/) into a lot of parts and then installing back proper TweakScale support.
